sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/V2ViewDescription.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.analysis
+
+import scala.collection.JavaConverters._
+
+import org.apache.spark.sql.catalyst.plans.logical.ViewDescription
+import org.apache.spark.sql.connector.catalog.{View, ViewCatalog}
+import org.apache.spark.sql.types.StructType
+
+/**
+ * View description backed by a View in V2 catalog.
+ *
+ * @param view a view in V2 catalog
+ */
+case class V2ViewDescription(
+    override val identifier: String,
+    view: View) extends ViewDescription {
+
+  override val schema: StructType = view.schema
+
+  override val viewText: Option[String] = None
```

Comment:
Isn't it supposed to be `Option(view.sql)`? Or how does it differ from the `val sql` below? I think we just need to keep one for callers?

Reply:
You are right! And `V2ViewDescription.sql` can be removed.

Reply:
How about changing `ShowCreateV2View` to take `(Identifier, View)` instead of `V2ViewDescription`?

Reply:
What is the benefit of doing such a modification? And by the way, I found `output` parameter in `ShowCreateViewExec(output: Seq[Attribute], desc: V2ViewDescription)` is never used? Maybe we can remove this parameter.

Reply:
`ShowCreateV2View` is for v2 view while `V2ViewDescription` is an adaptor between v1 and v2, so it makes more sense for `ShowCreateV2View` not to use `V2ViewDescription`, instead, relying on V2 interfaces solely.

Reply:
>  I found output parameter in ShowCreateViewExec(output: Seq[Attribute], desc: V2ViewDescription) is never used?

It is needed by the abstract class `QueryPlan`.

Reply:
Thanks for explaining. Do you think we also need to modify `ShowViewPropertiesExec` and `DescribeViewExec` which uses `V2ViewDescription`? Looks like using `V2ViewDescription` would bring some convenience especially for `DescribeViewExec`.

Reply:
Indeed, `V2ViewDescription` is an API adaptor for easy cloning of v1 SQL commands, and adding missing APIs like `comment` or `owner` (stored in properties on disk), etc.

So keep using `V2ViewDescription`?

Reply:
Yeah I am fine with keep using `V2ViewDescription`, I don't think the change is necessary.